packages/framework-core/src/booster-event-stream-producer.ts

```diff
@@ -0,0 +1,18 @@
+import { Trace } from './instrumentation'
+import { BoosterConfig, TraceActionTypes } from '@boostercloud/framework-types'
+import { getLogger } from '@boostercloud/framework-common-helpers'
+import { RawEventsParser } from './services/raw-events-parser'
+
+export class BoosterEventStreamProducer {
+  @Trace(TraceActionTypes.PRODUCE_STREAM_EVENTS)
+  public static async produce(request: unknown, config: BoosterConfig): Promise<void> {
+    const logger = getLogger(config, 'BoosterEventStreamProducer#produce')
+    logger.debug('Produce event workflow started for request:', require('util').inspect(request, false, null, false))
+    try {
+      const eventEnvelopes = config.provider.events.rawToEnvelopes(request)
+      await RawEventsParser.streamPerEntityEvents(config, eventEnvelopes, config.provider.events.produce)
```

Comment:
I still have a question here: Why do we need to produce the events grouped by entity name and ID (sharding key) if EventHub will already manage that grouping by itself?

My guess is that EventHub will automatically split events among partitions by sharding key